src/include/miopen/l1loss/solvers.hpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/l1loss/problem_description.hpp>
+#include <miopen/solver.hpp>
+
+#include <utility>
+
+namespace miopen {
+
+namespace solver {
+
+namespace l1loss {
+
+using L1LossForwardSolverBase =
+    NonTunableSolverBase<ExecutionContext, miopen::l1loss::L1LossFwdProblemDescription>;
+
+struct L1LossForward5d final : L1LossForwardSolverBase
+{
+    const std::string& SolverDbId() const override { return GetSolverDbId<L1LossForward5d>(); }
+
+    bool IsApplicable(const ExecutionContext& context,
+                      const miopen::l1loss::L1LossFwdProblemDescription& problem) const override;
+    ConvSolution
+    GetSolution(const ExecutionContext& context,
+                const miopen::l1loss::L1LossFwdProblemDescription& problem) const override;
+    std::size_t
+    GetWorkspaceSize(const ExecutionContext& context,
+                     const miopen::l1loss::L1LossFwdProblemDescription& problem) const override;
+    bool MayNeedWorkspace() const override { return true; }
```

Comment:
Because I can't retrieve 'reduction' variable at this stage so I handle the case at l1loss solver
```
std::size_t
L1LossForward5d::GetWorkspaceSize(const ExecutionContext& /*context*/,
                                  const miopen::l1loss::L1LossFwdProblemDescription& problem) const
{
    if(problem.GetReduction() == MIOPEN_L1LOSS_NONE_REDUCTION)
    {
        return 0;
    }

    return (problem.GetIDesc().GetElementSize() +
            AlignUp(problem.GetIDesc().GetElementSize(), LOCAL_SIZE_REDUCE_FWD) /
                LOCAL_SIZE_REDUCE_FWD) *
           get_data_size(problem.GetODesc().GetType());
}
```

Reply:
you are right!